File: concrete/src/test/java/pcb/uwu/amount/composite/electromagnetism/WebbersTest.java
package pcb.uwu.amount.composite.electromagnetism;

import org.junit.Test;
import pcb.uwu.amount.composite.fundamental.Area;
import pcb.uwu.amount.composite.termodynamics.Joules;
import pcb.uwu.amount.quantity.Amperes;
import pcb.uwu.amount.quantity.Seconds;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.core.Magnitude.KILO;
import static pcb.uwu.core.Magnitude.MILLI;
import static pcb.uwu.unit.composite.fundamental.SquareMeterUnit.SQUARE_METER;

public class WebbersTest {
	private static final Webbers webbers = new Webbers(12);

	@Test
	public void dividedByTime() {
		assertEquals(new Volts(2),
				webbers.dividedBy(new Seconds(6), DECIMAL64));
	}

	@Test
	public void dividedByElectricPotential() {
		assertEquals(new Seconds(6),
				webbers.dividedBy(new Volts(2), DECIMAL64));
	}

	@Test
	public void dividedByElectricCurrent() {
		assertEquals(new Henries(2, KILO),
				webbers.dividedBy(new Amperes(6, MILLI), DECIMAL64));
	}

	@Test
	public void dividedByElectricInductance() {
		assertEquals(new Amperes(6, MILLI),
				webbers.dividedBy(new Henries(2, KILO), DECIMAL64));
	}

	@Test
	public void dividedByMagneticField() {
		assertEquals(new Area(2, SQUARE_METER),
				webbers.dividedBy(new Teslas(6), DECIMAL64));
	}

	@Test
	public void dividedByArea() {
		assertEquals(new Teslas(6),
				webbers.dividedBy(new Area(2, SQUARE_METER), DECIMAL64));
	}

	@Test
	public void multiplyByElectricCurrent() {
		assertEquals(new Joules(600, MILLI),
				webbers.multipliedBy(new Amperes(50, MILLI), DECIMAL64));
	}

}

File: implementation/src/test/java/pcb/uwu/amount/derived/electromagnetism/WebbersTest.kt
package pcb.uwu.amount.derived.electromagnetism

import pcb.uwu.amount.base.Amperes
import pcb.uwu.amount.base.Seconds
import pcb.uwu.amount.derived.geometry.Area
import pcb.uwu.amount.derived.geometry.area.SquareMeters
import pcb.uwu.amount.derived.thermodynamics.Joules
import pcb.uwu.core.Magnitude.KILO
import pcb.uwu.core.Magnitude.MILLI
import pcb.uwu.unit.derived.geometry.SquareMeterUnit
import kotlin.test.Test
import kotlin.test.assertEquals

class WebbersTest
{
    @Test
    fun dividedByTime() =
        assertEquals(expected = Volts(2),
                     actual = Webbers(12) / Seconds(6))

    @Test
    fun dividedByElectricPotential() =
        assertEquals(expected = Seconds(6),
                     actual = Webbers(12) / Volts(2))

    @Test
    fun dividedByElectricCurrent() =
        assertEquals(expected = Henries(2000),
                     actual = Webbers(12) / Amperes(6, MILLI))

    @Test
    fun dividedByElectricInductance() =
        assertEquals(expected = Amperes(0.006),
                     actual = Webbers(12) / Henries(2, KILO))

    @Test
    fun dividedByMagneticField() =
        assertEquals(expected = SquareMeters(2),
                     actual = Webbers(12) / Teslas(6))

    @Test
    fun dividedByArea() =
        assertEquals(expected = Teslas(6),
                     actual = Webbers(12) / Area(amount = 2, unit = SquareMeterUnit))

    @Test
    fun multiplyByElectricCurrent() =
        assertEquals(expected = Joules(0.6),
                     actual = Webbers(12) * Amperes(50, MILLI))

}